### DISCUSSION TITLE
Resuming an experiment mid-session

### DISCUSSION THREAD

**Discussion by jabeim** [jspsych/jsPsych#2831] [Q&A]
I'm currently working on building a JsPsych experiment that I would like to host on my own server and would like to enable visitors to partially complete an experiment and resume it later.

My current design saves trial data to a database at the end of each trial, and I'd like to be able to somehow tell jsPsych to not start at the first timeline node but instead the one after the most recently completed one.

This seems easy enough if I start by counting the completed trials and removing the same number from the front of the timeline before calling jsPsych.run() but it quickly gets complicated when trying to support nested timelines, repetitions, and randomization.

Does anyone have examples or experience with enabling this type of functionality for their own purposes?

**Comment by jodeleeuw** [comment]
hi @jabeim, I don't have any great advice about how to do this, but it sounds like something we should be thinking about how to support better with v8.0.

@bjoluc 👀 

In the meantime, maybe using the relatively new `setSeed()` feature ([docs here](https://www.jspsych.org/7.3/reference/jspsych-randomization/#jspsychrandomizationsetseed)) plus [simulation mode](https://www.jspsych.org/7.3/overview/simulation/) to simulate the previously completed trials could get you closer?
